pyccel/codegen/printing/ccode.py

```diff
@@ -55,6 +56,105 @@
     "ceiling": "ceil",
 }
 
+# dictionary mapping numpy function to (argument_conditions, C_function).
+# Used in CCodePrinter._print_NumpyUfuncBase(self, expr)
+numpy_ufunc_to_c = {
+    'NumpyAbs'  : 'fabs',
+    'NumpyMin'  : 'minval',
+    'NumpyMax'  : 'maxval',
+    'NumpyFloor': 'floor',  # TODO: might require special treatment with casting
+    # ---
+    'NumpyExp' : 'exp',
+    'NumpyLog' : 'log',
+    # 'NumpySqrt': 'sqrt',  # sqrt is printed using _Print_NumpySqrt
+    # ---
+    'NumpySin'    : 'sin',
+    'NumpyCos'    : 'cos',
+    'NumpyTan'    : 'tan',
+    'NumpyArcsin' : 'asin',
+    'NumpyArccos' : 'acos',
+    'NumpyArctan' : 'atan',
+    'NumpyArctan2': 'atan2',
+    'NumpySinh'   : 'sinh',
+    'NumpyCosh'   : 'cosh',
+    'NumpyTanh'   : 'tanh',
+    'NumpyArcsinh': 'asinh',
+    'NumpyArccosh': 'acosh',
+    'NumpyArctanh': 'atanh',
```

Comment:
Several of these functions have different calls for complex numbers (see https://en.wikibooks.org/wiki/C_Programming/complex.h). It may be worth having 2 dictionaries. One to handle real numbers and one to handle complexes